nbwidgets / fdfitting: update documentation (#478)
lumicks/pylake: docs/tutorial/nbwidgets.rst

Channel slicing
---------------

Let's say we want to do some analyses on slices of channel data.
It would be nice to just quickly visually select some regions using a widget.

> Let's load the file and open the :class:`~lumicks.pylake.nb_widgets.range_selector.SliceRangeSelectorWidget` which we can conveniently open for a slice using the :meth:`~lumicks.pylake.channel.Slice.range_selector()` method::

Reviewer: This sentence sounds a bit odd. Additionally we don't want to give the impression that the user should construct the widget manually (it's not public through `lk`) Maybe change to
```suggestion
Let's load the file and open an interactive plot using the :meth:`~lumicks.pylake.channel.Slice.range_selector()` method::
```

and then below the codeblock/figure can add
```
This returns a :class:`~lumicks.pylake.nb_widgets.range_selector.SliceRangeSelectorWidget` ....
```

Author: Yeah, that flows better.

    original.f.plot()
    sliced.f.plot(start=original.start)

    plt.tight_layout()

.. image:: figures/nbwidgets/fd_dist_widget2.png

The returned F,d curves correspond to the longest contiguous (in time) stretch of data that falls
within the distance thresholds.

Range selection of single curve
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The selector widgets can also be easily accessed from single :class:`~lumicks.pylake.fdcurve.FdCurve` instances using :meth:`~lumicks.pylake.fdcurve.FdCurve.range_selector`::

    plt.figure()
    fdcurve = fdcurves["40"]
    t_selector = fdcurve.range_selector()

.. image:: figures/nbwidgets/single_curve_widget1.png

And for the distance-based selector we can use :meth:`~lumicks.pylake.fdcurve.FdCurve.distance_range_selector`::

    d_selector = fdcurve.distance_range_selector()

.. image:: figures/nbwidgets/single_curve_widget2.png

.. _crop_and_rotate:

Cropping and Rotating Image Stacks
----------------------------------

> You can interactively define the location of a tether for a :class:`~lumicks.pylake.ImageStack` by using :meth:`~lumicks.pylake.ImageStack.crop_and_rotate`::

Reviewer: Thanks for adding the return type to the docstring here; however the link to the class doesn't work. Just use the class name (not the RST :class: link)

Additionally could you copy/paste the `Actions` section from `crop_and_rotate` to `ImageEditorWidget` 🙏

Author: Unfortunately, it only works if you add it to the listing (which I just did).
